pkg/api/network/store.go

```diff
@@ -79,3 +82,16 @@ func (s *networkStore) checkUniqueVlanID(ns string, config *NetConf) error {
 	}
 	return nil
 }
+
+func (s *networkStore) Delete(request *types.APIRequest, schema *types.APISchema, id string) (types.APIObject, error) {
+	vms, err := s.vmCache.GetByIndex(indexeres.VMByNetworkIndex, request.Name)
+	if err != nil {
+		return types.APIObject{}, apierror.NewAPIError(validation.ServerError, err.Error())
+	}
+
+	if len(vms) > 0 {
+		return types.APIObject{}, apierror.NewAPIError(validation.ServerError, "network is still in use")
```

Comment:
It seems better to use 400 Bad Request as the status code. Also, it would be helpful to know which VM is using the network.